### PR TITLE
Refactor input_type code into classes

### PR DESCRIPTION
**Pull request by ramaro**
Introduces a new InputType class which serves as the base class for 2 new input_type compilers:
* Jsonnet
* Jinja2

Also fixes minor code style issues

### CODE REVIEW

File: kapitan/inputs/base.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        if not file_found:
+            raise CompileError("Compile error: {} for target: {} not found in "
+                               "search_paths: {}".format(input_path, target_name, self.search_paths))
+
+    def make_compile_dirs(self, target_name, output_path):
+        """make compile dirs, skips if dirs exist"""
+        _compile_path = os.path.join(self.compile_path, target_name, output_path)
+        # support writing to an already existent dir
+        try:
+            os.makedirs(_compile_path)
```
Comment (uberspot) [inline kapitan/inputs/base.py]: Use `os.makedirs(_compile_path, exist_ok=True)` then you can remove the try-catch block you have in case it fails.

Comment (ramaro) [reply kapitan/inputs/base.py]: Good point. Done

File: kapitan/inputs/jinja2.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        # set ext_vars and inventory for jinja2 context
+        context = ext_vars.copy()
+        context["inventory"] = inventory(self.search_paths, target_name)
+        context["inventory_global"] = inventory(self.search_paths, None)
+
+        for item_key, item_value in render_jinja2(file_path, context).items():
+            full_item_path = os.path.join(compile_path, item_key)
+            try:
+                os.makedirs(os.path.dirname(full_item_path))
```
Comment (uberspot) [inline kapitan/inputs/jinja2.py]: Same here, `exist_ok=True`

File: kapitan/inputs/jsonnet.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                with CompiledFile(file_path, self.ref_controller, mode="w", reveal=reveal, target_name=target_name,
+                                  indent=indent) as fp:
+                    fp.write_json(item_value)
+                    logger.debug("Wrote %s", file_path)
+            elif output == 'yaml':
+                file_path = os.path.join(compile_path, '%s.%s' % (item_key, "yml"))
+                with CompiledFile(file_path, self.ref_controller, mode="w", reveal=reveal, target_name=target_name,
+                                  indent=indent) as fp:
+                    fp.write_yaml(item_value)
+                    logger.debug("Wrote %s", file_path)
```
Comment (uberspot) [inline kapitan/inputs/jsonnet.py]: Would it make sense to move this `logger.debug` call in a write_yaml super() function in CompiledFile ? Then you don't have to explicitly add it everywhere again and you always have a debug log.

File: kapitan/inputs/jinja2.py (hunk 2 above)
Comment (ramaro) [reply kapitan/inputs/jinja2.py]: done

File: kapitan/targets.py [hunk 4]
```diff
@@ -288,145 +287,27 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, **kwa
...
-        with CompiledFile(full_item_path, ref_controller, mode="w", reveal=reveal, target_name=target_name) as fp:
-            fp.write(item_value["content"])
-            mode = item_value["mode"]
-            os.chmod(full_item_path, mode)
-            logger.debug("Wrote %s with mode %.4o", full_item_path, mode)
+            input_compiler = jinja2_compiler
+        elif input_type == "jsonnet":
+            input_compiler = jsonnet_compiler
+        else:
+            raise CompileError("")
```
Comment (adrianchifor) [inline kapitan/targets.py]: We should print that the input type is not valid, and which ones we support

Comment (ramaro) [reply kapitan/targets.py]: Oops, forgot that. Will fix.

File: kapitan/inputs/jsonnet.py (hunk 3 above)
Comment (ramaro) [reply kapitan/inputs/jsonnet.py]: Yep

Comment (ramaro) [reply kapitan/inputs/jsonnet.py]: done

File: kapitan/targets.py (hunk 4 above)
Comment (ramaro) [reply kapitan/targets.py]: done

**Pull request merged**